src/components/Layout/Header/ActionCenter/components/Notifications/SwapNotification.tsx

```diff
@@ -0,0 +1,127 @@
+import { CloseIcon } from '@chakra-ui/icons'
+import { Box, Button, Flex, HStack, Icon, Stack } from '@chakra-ui/react'
+import type { RenderProps } from '@chakra-ui/react/dist/types/toast/toast.types'
+import { fromBaseUnit } from '@shapeshiftoss/utils'
+import { useMemo } from 'react'
+import { TbArrowRight, TbCircleCheckFilled, TbCircleXFilled } from 'react-icons/tb'
+
+import { Amount } from '@/components/Amount/Amount'
+import { AssetIcon } from '@/components/AssetIcon'
+import { StreamIcon } from '@/components/Icons/Stream'
+import { SwapperIcon } from '@/components/MultiHopTrade/components/TradeInput/components/SwapperIcon/SwapperIcon'
+import { swapSlice } from '@/state/slices/swapSlice/swapSlice'
+import { useAppSelector } from '@/state/store'
+
+type SwapNotificationProps = {
+  handleClick: () => void
+} & RenderProps
+
+const toastHoverProps = {
+  transform: 'scale(1.02)',
+}
+
+export const SwapNotification = ({ handleClick, status, id, onClose }: SwapNotificationProps) => {
+  const swapsById = useAppSelector(swapSlice.selectors.selectSwapsById)
+
+  const swap = useMemo(() => {
+    if (!id) return undefined
+    return swapsById[id]
+  }, [id, swapsById])
```

Comment:
`id` should be defined, see https://github.com/shapeshift/web/pull/9695/files#r2150415634

Reply:
it's only optional at types level by virtue of us leveraging chakra types, see https://github.com/shapeshift/web/pull/9695#discussion_r2152039535. We always call swap action toasts with a defined ID:

<img width="405" alt="image" src="https://github.com/user-attachments/assets/4e8f6d31-032b-4059-86fe-019a02d74451" />